### PR TITLE
Track encoded boxes in the RetinaNet in a Keras metric

### PR DESCRIPTION
**Pull request by LukeWood**
fixes https://github.com/keras-team/keras-cv/issues/855

### CODE REVIEW

**Review by tanzhenyu — COMMENTED** [review]
I wonder whether we should do this through a callback, instead of impacting the API?

**Comment by LukeWood** [comment]
> I wonder whether we should do this through a callback, instead of impacting the API?

Good point.  I feel this actually relates to the PyMetric in a way; in particular we want to add a metric to each epoch outside of the train/test step.

File: keras_cv/layers/object_detection/retina_net_label_encoder.py [hunk 1]
```diff
@@ -36,6 +36,11 @@ class RetinaNetLabelEncoder(layers.Layer):
         background_class: (Optional) The class ID used for the background class.
             Defaults to -1.
         ignore_class: (Optional) The class ID used for the ignore class. Defaults to -2.
+        track_matched_boxes: (Optional) Whether or not to track the number of boxes that
```
Comment (tanzhenyu) [inline keras_cv/layers/object_detection/retina_net_label_encoder.py:39]: discussed offline -- let's just turn on this without impacting API

Comment (LukeWood) [reply keras_cv/layers/object_detection/retina_net_label_encoder.py:39]: (This is in the encoder - RetinaNet still doesn't have a way to disable)

Comment (LukeWood) [reply keras_cv/layers/object_detection/retina_net_label_encoder.py]: Updated

**Review by tanzhenyu — APPROVED** [review]

**Pull request merged**